md2conf/converter.py

```diff
@@ -297,10 +310,11 @@ def _transform_link(self, anchor: ET._Element) -> None:
         )
         self.links.append(url)
 
+        tiny_link_id: str = page_id_to_tiny_link(int(link_metadata.page_id))
         components = ParseResult(
             scheme="https",
             netloc=link_metadata.domain,
-            path=f"{link_metadata.base_path}spaces/{link_metadata.space_key}/pages/{link_metadata.page_id}/{link_metadata.title}",
+            path=f"{link_metadata.base_path}x/{tiny_link_id}",
```

Comment:
I am afraid the logic may be more complex than this.

If I follow the instructions on the page [How to programmatically generate the tiny link of a Confluence page](https://confluence.atlassian.com/confkb/how-to-programmatically-generate-the-tiny-link-of-a-confluence-page-956713432.html), and click the *Share* button to generate a tiny URL, then I get a very different URL in my Confluence environment than what the function `page_id_to_tiny_link` returns. In fact, the generated URL is different each time I load the page and click *Share*, so I am assuming a lookup table is being used.

I would also make tiny links a configurable setting and not force this on all users, since tiny URLs appear to be a redirection feature, and the longer structured URLs seem to be the standard way that Confluence uses for routing.